### PR TITLE
TC Custom Service Importer

### PR DESCRIPTION
**Pull request by awisema**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Importer for custom services. As the exporter has not yet been built, there are probably some adjustments to be made to make it conform to the export.

## Type of change
[//]: # 'remove options that are not relevant'
- [X] New feature (adds functionality)

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [X] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Review by ttoomey — APPROVED** [review]
Looks good, thanks! I'll take care of handling the merge

File: drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb]: Awesome, would be great to normalize into standard values. From what I saw it is often just the casing and punctuation that differed across services, but it needs a closer look.

Suggestion :

Normalized Service Location options:
* Place not meant for habitation
* Service Setting, Non Residential
* Service Setting, Residential

Normalized Type of Contact options:
* Home
* Phone
* Phone Left Voice Message
* Email
* Office
* In-Person
* Driving Client
* No Show Home Visit
* No Contact/No Show
* Counseling
* Case Management Advocacy
* Other

File: drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb:125]: Services that collect financial assistance amount should store it in this field, _not_ as a custom data element. I think here you're storing it in data elements with key "FAAmount", which we don't want

File: drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb:233]: See note above, this can be stored directly on the CustomService table

**Comment by gigxz** [comment]
thank you Sandy this looks great!!!

**Comment by ttoomey** [comment]
@gigxz do you think this PR ready to merge? If so I can fix up the conflicts and merge it in

File: drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb
Comment (gigxz) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb:125]: nevermind, I see they are nested under `service_fields`

**Comment by gigxz** [comment]
@ttoomey I am going to open a PR against this branch with a few suggestions

**Comment by gigxz** [comment]
alllright @ttoomey if you can merge in https://github.com/greenriver/hmis-warehouse/pull/4093 and then resolve merge conflicts and merge to `release-107` that would be awesome!

Hopefully I caught everything in that PR. I did an export of the last few months and compared the field names. (A few hadn't been used in the last few months so I needed to run a longer history to look at those: Baby Supplies; Benefits Contacts; Budgeting/FInancial Planning; and Food Box ).

File: drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb
Comment (ttoomey) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb:104]: @gigxz something to flag: I exported everything from 2023 to present (selecting just the touchpoints used in this importer) and it looks like about 50% of the records had a blank enrollment_id

**Comment by ttoomey** [comment]
Fixes:
* retarget this to release-108
* use only the first line from Question
* create missing service types in development mode
* don't invoke cleaner if the answer value is empty
* clean curly braces out of enrollment_id
* fix service type name s/1A-SA Dinner/1-SA Dinner/

**Comment by ttoomey** [comment]
@gigxz I made a few additional changes, could use a quick review before merge. Thanks!

File: drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb
Comment (gigxz) [reply drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_services_loader.rb:104]: @ttoomey thank you - that is good to know - I will flag to them

**Review by gigxz — APPROVED** [review]
looks good to me! thanks

**Pull request merged**
